typescript/schedules.md

```diff
@@ -0,0 +1,1257 @@
+   * takes those Actions according to the {@link ScheduleOverlapPolicy}. An outage that lasts longer than the Catchup
+   * Window could lead to missed Actions. (But you can always {@link ScheduleHandle.backfill}.)
+   *
+   * @default 1 minute
+   */
+  catchupWindow: Milliseconds;
+
+  /**
+   * When an Action times out or reaches the end of its Retry Policy, {@link pause}.
+   *
+   * With {@link ScheduleOverlapPolicy.AllowAll}, this pause might not apply to the next Action, because the next Action
+   * might have already started previous to the failed one finishing. Pausing applies only to Actions that are scheduled
+   * to start after the failed one finishes.
+   *
+   * @default false
+   */
+  pauseOnFailure: boolean;
+
+  /**
+   * Informative human-readable message with contextual notes, e.g. the reason
+   * a Schedule is paused. The system may overwrite this message on certain
+   * conditions, e.g. when pause-on-failure happens.
+   */
+  note?: string;
+
+  /**
+   * Is currently paused.
+   *
+   * @default false
+   */
+  paused: boolean;
+
+  /** Whether the number of Actions to take is limited. */
+  actionsAreLimited: boolean;
+
+  /**
+   * Limit the number of Actions to take.
+   *
+   * This number is decremented after each Action is taken, and Actions are not
+   * taken when the number is `0` (unless {@link ScheduleHandle.trigger} is called).
+   *
+   * @default unlimited
+   */
+  remainingActions?: NonNegativeInteger;
+}
+
+export interface ScheduleListPage {
+  schedules: Schedule[];
+
+  /**
+   * Token to use to when calling {@link ScheduleClient.listByPage}.
+   *
+   * If `undefined`, there are no more Schedules.
+   */
+  nextPageToken?: string;
+}
+
+export type WorkflowExecution = Required<temporal.api.common.v1.IWorkflowExecution>;
+
+export interface ScheduleInfo {
+  /** Number of Actions taken so far. */
+  numActionsTaken: number;
+  // TODO or numberOfActionsTaken etc?
+
+  /** Number of times a scheduled Action was skipped due to missing the catchup window. */
+  numActionsMissedCatchupWindow: number;
+
+  /** Number of Actions skipped due to overlap. */
+  numActionsSkippedOverlap: number;
+
+  /**
+   * Currently-running workflows started by this schedule. (There might be
+   * more than one if the overlap policy allows overlaps.)
+   * Note that the run_ids in here are the original execution run ids as
+   * started by the schedule. If the workflows retried, did continue-as-new,
+   * or were reset, they might still be running but with a different run_id.
+   */
+  runningWorkflows: WorkflowExecution[];
+
+  /**
+   * Most recent 10 Actions started (including manual triggers).
+   *
+   * Sorted from older start time to newer.
+   */
+  recentActions: ScheduleAction[];
+
+  /** Next 10 scheduled Action times */
+  nextActionTimes: Date[];
+
+  createdAt: Date;
+  lastUpdatedAt: Date;
+
+  isValid: boolean;
```

Comment:
What's `isValid`?

Reply:
`info.isValid` is a getter method that's more readable than `!info.invalidScheduleError`

Reply:
Right, now I remember I commented on this...
Still would like to understand why a schedule would be invalid in the first place.